servers/jax-rs-tests/src/test/java/org/projectnessie/jaxrs/AbstractRestSecurityContext.java

```diff
@@ -103,32 +98,46 @@ public void committerAndAuthor(
                   .contains("with security")
                   .contains("no security context");
             });
+  }
 
-    // Merge (individual commits)
+  @Test
+  public void committerAndAuthorMergeUnsquashed(
+      @NessieSecurityContext Consumer<SecurityContext> secContext) throws Exception {
+    Branch main = makeCommits(createBranch("committerAndAuthorMergeUnsquashed_main"), secContext);
+    Branch merge = createBranch("committerAndAuthorMergeUnsquashed_target");
 
-    getApi()
-        .mergeRefIntoBranch()
-        .fromRef(withSecurityContext)
-        .branch(merge2)
-        .keepIndividualCommits(true)
-        .merge();
+    secContext.accept(PrincipalSecurityContext.forName("NessieHerself"));
+
+    getApi().mergeRefIntoBranch().fromRef(main).branch(merge).keepIndividualCommits(true).merge();
 
-    merge2 = (Branch) getApi().getReference().refName(merge2.getName()).get();
+    merge = (Branch) getApi().getReference().refName(merge.getName()).get();
 
-    assertThat(getApi().getCommitLog().reference(merge2).maxRecords(2).get().getLogEntries())
+    assertThat(getApi().getCommitLog().reference(merge).maxRecords(2).get().getLogEntries())
         .extracting(LogEntry::getCommitMeta)
         .extracting(CommitMeta::getCommitter, CommitMeta::getAuthor, CommitMeta::getMessage)
         .containsExactly(
             tuple("NessieHerself", "ThatNessieGuy", "with security"),
             tuple("NessieHerself", "", "no security context"));
+  }
+
+  @Test
+  public void committerAndAuthorTransplant(
+      @NessieSecurityContext Consumer<SecurityContext> secContext) throws Exception {
+    Branch main = makeCommits(createBranch("committerAndAuthorTransplant_main"), secContext);
+    Branch transplant = createBranch("committerAndAuthorTransplant_target");
+
+    List<String> hashesToTransplant =
+        getApi().getCommitLog().reference(main).maxRecords(2).get().getLogEntries().stream()
+            .map(logEntry -> logEntry.getCommitMeta().getHash())
+            .collect(Collectors.toList());
+    Collections.reverse(hashesToTransplant); // transplant in the original commit order
 
-    // Transplant
+    secContext.accept(PrincipalSecurityContext.forName("NessieHerself"));
 
     getApi()
         .transplantCommitsIntoBranch()
-        .fromRefName(withSecurityContext.getName())
-        .hashesToTransplant(
-            Arrays.asList(noSecurityContext.getHash(), withSecurityContext.getHash()))
+        .fromRefName(main.getName())
+        .hashesToTransplant(hashesToTransplant)
         .branch(transplant)
         .keepIndividualCommits(true)
```

Comment:
So, this test case name should be `committerAndAuthorTransplantUnsquashed`?

Also, are we missing the squashed `committerAndAuthorTransplant` test case?

Reply:
Transplant is always unsquashed, AFAIK